packages/compile-solidity/src/index.js

```diff
@@ -182,7 +182,17 @@ const Compile = {
   },
 
   async sourcesWithPragmaAnalysis({ paths, options }) {
-    return compileWithPragmaAnalysis({ paths, options });
+    const compilationResult = await compileWithPragmaAnalysis({
+      paths,
+      options
+    });
+    return compilationResult.compilations.every(
```

Comment:
Is there a reason for the switch here?  I don't see any reason not to just promote whatever you get.

Reply:
Oh, I kind of assumed the empty one was a special case and this might be necessary. Happy to remove this!